azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"connectivity_endpoints": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"location": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"name": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"port": {
+							Computed: true,
+							Type:     schema.TypeInt,
+						},
+						"protocol": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+			"location": locationSchema(),
+			"name": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"os_type": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Linux",
+					"Windows",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+			"resource_group_name": resourceGroupNameSchema(),
+			"restauth_username": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"restauth_password": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"head_node": {
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"linux_os_profile": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"password": {
+										Optional: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"ssh_keys": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeList,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"key_data": {
+													Required: true,
+													ForceNew: true,
+													Type:     schema.TypeString,
+												},
+											},
+										},
+									},
+									"username": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+								},
+							},
+						},
+						"min_instance_count": {
+							Optional:     true,
+							Default:      1,
+							ForceNew:     true,
+							Type:         schema.TypeInt,
+							ValidateFunc: validation.IntBetween(1, 32),
+						},
+						"script_actions": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"parameters": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
```

Comment:
shouldn't this be a TypeMap which gets stitched together?

Reply:
There's no difference for a user to pass it in a format of map or just a string. And sometimes a string would seems to be more intuitive (like `-DartifactName="Abcde" --enable-debug true`).